### PR TITLE
EZP-31575: When in Media section show Sub-Items grid view by default

### PR DESCRIPTION
**Pull request by lucasOsti**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-31575
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | no
| Doc needed?   | yes/no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Set grid view of sub items for media site

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/admin.location.view.js [hunk 1]
```diff
@@ -129,6 +129,7 @@
     };
 
     listContainers.forEach((container) => {
+        const activeView = container.dataset.location == eZ.adminUiConfig.locations.media ? 'grid' : 'table';
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/admin.location.view.js:132]:
```suggestion
        const activeView = container.dataset.location === eZ.adminUiConfig.locations.media ? 'grid' : 'table';
```

File: src/bundle/Resources/public/js/scripts/admin.location.view.js [hunk 2]
```diff
@@ -154,7 +156,8 @@
             React.createElement(eZ.modules.SubItems, {
                 handleEditItem,
                 generateLink,
-                parentLocationId: parseInt(container.dataset.location, 10),
+                activeView,
+                parentLocationId: parentLocationId,
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/admin.location.view.js:160]:
```suggestion
                parentLocationId,
```

File: src/bundle/Resources/public/js/scripts/admin.location.view.js [hunk 3]
```diff
@@ -150,11 +159,14 @@
             return contentTypeDataMap;
         }, {});
 
+        console.log('View: ', activeView);
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/admin.location.view.js:162]:
```suggestion
```

File: src/bundle/Resources/public/js/scripts/admin.location.view.js [hunk 4]
```diff
@@ -127,8 +127,17 @@
 
         $(SELECTOR_MODAL_BULK_ACTION_FAIL).modal('show');
     };
+    const getLocationActiveView = (parentLocationId) => {
+        const mediaLocationId = eZ.adminUiConfig.locations.media;
+        const defaultActiveView = parentLocationId === mediaLocationId ? 'grid' : 'table';
+        const activeView = localStorage.getItem(`ez-location-active-view-${parentLocationId}`);
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/admin.location.view.js:133]: I would name it `ez-subitems-active-view-location`

File: src/lib/UI/Config/Provider/Locations.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        $this->configResolver = $configResolver;
+    }
+
+    public function getConfig(): array
+    {
+        return [
+            self::MEDIA_IDENTIFIER => $this->configResolver->getParameter('location_ids.media'),
+            self::CONTENT_STRUCTURE_IDENTIFIER => $this->configResolver->getParameter('location_ids.content_structure'),
+            self::USERS_IDENTIFIER => $this->configResolver->getParameter('location_ids.users'),
+        ];
```
Comment (adamwojs) [inline src/lib/UI/Config/Provider/Locations.php:35]: Q: Do you need entries for a Form Builder or Site Factory skeletons?

**Pull request merged**